build-tools/packages/build-tools/src/test/biomeConfig.test.ts

```diff
@@ -0,0 +1,214 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "node:assert/strict";
+import path from "node:path";
+import {
+	getBiomeFormattedFiles,
+	getSettingValuesFromBiomeConfig,
+	loadBiomeConfig,
+} from "../common/biomeConfig";
+import type { Configuration as BiomeConfig } from "../common/biomeConfigTypes";
+import { getResolvedFluidRoot } from "../common/fluidUtils";
+import { GitRepo } from "../common/gitRepo";
+import { testDataPath } from "./init";
+
+describe("BiomeConfig", async () => {
+	describe("loadConfig", async () => {
+		it("throws on missing config", async () => {
+			const testFile = path.resolve(testDataPath, "biome/missing.jsonc");
+			assert.rejects(async () => await loadBiomeConfig(testFile), Error);
+		});
+
+		it("throws on empty config", async () => {
+			const testFile = path.resolve(testDataPath, "biome/empty.jsonc");
+			assert.rejects(async () => await loadBiomeConfig(testFile), Error);
+		});
+
+		it("loads single config", async () => {
+			const testFile = path.resolve(testDataPath, "biome/base.jsonc");
+			const actual = await loadBiomeConfig(testFile);
+			assert.notEqual(actual, undefined);
+			assert.equal(actual.files?.ignoreUnknown, true);
+		});
+
+		it("loads config with multiple extends", async () => {
+			const testFile = path.resolve(testDataPath, "biome/pkg-b/biome.jsonc");
+			const actual = await loadBiomeConfig(testFile);
+
+			assert(actual !== undefined);
+			assert(actual.files?.ignoreUnknown === false);
+
+			assert(actual.files?.include?.includes("pkg-a-include/**"));
+			assert(actual.files?.ignore?.includes("pkg-a-ignore/**"));
+			assert(actual.formatter?.include?.includes("include-formatter-added-1/**"));
+			assert(actual.formatter?.ignore?.includes("ignore-formatter-added-1/**"));
+			assert(actual.linter?.include?.includes("include-linter-added-1/**"));
+			assert(actual.linter?.ignore?.includes("ignore-linter-added-1/**"));
```

Comment:
I feel like you want to use `!` for these instead of `?`, because there is no case where it is okay for those things to be missing/undefined. (`?` usually means "okay, I'll handle it")

Reply:
I didn't do that because of @typescript-eslint/no-non-null-assertion. Is it worth suppressing?

Reply:
And I assume you're referring to all the uses in this file - or are these special in some way?

Reply:
I am referring to all. `no-non-null-assertion` is a pretty aggressive lint rule. I find myself turning it off a lot.
The lint rule is hey don't use this language feature that has a real use case.
Since these don't match the regular pattern for `?`, I would go with `!` and suppress that check for the whole file.